### DISCUSSION TITLE
Maximum IR range?

### DISCUSSION THREAD

**Discussion by OguzCiftci** [Arduino-IRremote/Arduino-IRremote#977] [Q&A] [answered]
Hi, may I ask what is the maximum range you can get by using this library with 3.3V supply? I use TSAL6400 IR LED connecting 20-50 ohm resistor in series which is controlled with BC547 and the maximum range I got is around 3 meters.(LED is supplied from a regulator directly so there is almost no limit for current) Other than reflectors and LED arrays, is there any other suggestions to improve the range further at 3.3V?

I use a Samsung TV for testing.

Thanks

**Comment by ArminJo** [comment, marked as answer]
Look at the IR Led datasheet and choose a lower resistor, or choose a IR led with a narrow angle.

**Reply by OguzCiftci** [reply]
I tried both, they don't make much difference. The range difference between TV remote and the setup I use (includes ESP32) is almost doubled. Remote uses the same IR led but uses ADAM27P16G to control. I tried to run IR LED at 150mA (max 200mA) no luck without increasing the supply voltage. Any suggestion is much appreciated, thank you.

**Reply by ArminJo** [reply]
How do ou know you have 200 mA? Maybe the transistor base resistor is too high for totally switching it.

**Reply by OguzCiftci** [reply]
I used an external supply to measure the current at the peak and set the resistor value according to that then, lowered the resistor value to get the max current I can get but, you have a point, when the code runs, it will not stay at the peak value. I will try it with different transistors and resistor values, thank you.

I will mark this answered because, I will only be able to try this new setup in middle of the next week, I will let you know about the result. Thanks.

**Reply by ArminJo** [reply]
We send with a duty cycle of 30%, so peak current can be up to 600 mA.

**Reply by AnalysIR** [reply]
To measure your IR LED current:

1.  Write a new sketch without IRremote
2.  Set the IR send GPIO pin  to OUTPUT+HIGH for a few seconds (~5 secs) ONLY, to avoid stressing the LED
3.  Measure the voltage Vir across the emitter resistor Re (or resistor in line with IR LED)
4. The IR Tx current will be Vir/Re * 1000 in mA

To try again just reset the Arduino and get another 5 secs.

**Reply by ArminJo** [reply]
Or generate a tone at this output, then you have a duty cycle of 50% and do not stress your diode so much. You must then use the double of your measured Vir.

**Reply by OguzCiftci** [reply]
Idk if I do something wrong but when I set the duty cycle from 30 to 130, the distance is tripled. I didn't have any chance to measure anything technically yet. I get around 15 meters with 5V supply which is more than enough of what I need. Thank you both.

**Reply by AnalysIR** [reply]
> when I set the duty cycle from 30 to 130

Not sure what would happen inside the code, with a 130% duty cycle configured (which is not really possible). Stick to 50% or 33.3%, unless you have well informed reasons to pick another value.

If you are running on battery with low power usage as a priority, use 33.3% duty cycle and design the IR emitter circuit around that.
Otherwise, 50% or 33.3% duty cycle is the way to go. 50% will get you more range in the **same circuit design**. (33.3% designs are more efficient, _in terms of range per mA_)

On some boards, the 3V3 supply has limited current available relative to the 5V supply. So I usually design the IR emitters to use the 5V supply (where possible) with the transistor driven by the 3V3 GPIO.

@5V....by placing 2 IR LEDs in series you essentially get double power for free! (again need to design the IR emitter circuit accordingly)
